Expensify/App
# Always auto name and create new workspaces from oldDot

@TomatoToaster pinging you for review since we discussed this PR.

### Details
Removes the `workspace/new` page completely since we always want to auto name and create workspaces for N6.
[Slack thread](https://expensify.slack.com/archives/C020EPP9B9A/p1628547819480000) for more context, but in N6 we always want to auto name and create the workspace for the user, so the `workspace/new` page is no longer needed.

Also removed the success growl per [this](https://github.com/Expensify/Expensify/issues/173338) issue.

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/173349
$ https://github.com/Expensify/Expensify/issues/173338

### Tests
Tested through oldDot Inbox flow.
1. Log into an account on oldDot and show all hidden inbox cards
2. Find the inbox card below and select only `Get Expensify Card` then click `Submit`
![image](https://user-images.githubusercontent.com/3387421/128937424-fcb96d9b-df7b-42ff-8c4d-a5a18326ae69.png)
3. Verify you are taken to `/workspace/<policyId>/card`
![image](https://user-images.githubusercontent.com/3387421/128937778-17e78653-8791-4d0e-acfd-561631188b11.png)

### QA Steps
Same as tests

### Tested On


#### Web
https://user-images.githubusercontent.com/3387421/128938529-9ca25e13-5942-4c79-8bd6-1807b4693b1e.mov
